src/pyenphase/envoy.py

```diff
@@ -441,6 +441,8 @@ async def update(self) -> EnvoyData:
                 await updater.update(data)
             except EndOfStream as err:
                 raise EnvoyCommunicationError("EndOfStream at update") from err
+            except (httpx.ConnectError, httpx.TimeoutException) as err:
+                raise EnvoyCommunicationError(f"{err}") from err
```

Comment:
Maybe we want a separate exception for timeouts?

Reply:
> Maybe we want a separate exception for timeouts?

probably best as the httpx.TimeoutException does not return any description.

Reply:
Split them and changed httpx.ConnectError to httpx.NetworkError to catch more general httpx network errors.